picasso/src/main/java/com/squareup/picasso/PicassoDrawable.java

```diff
@@ -109,10 +121,24 @@ static void setPlaceholder(ImageView target, int placeholderResId, Drawable plac
           placeholder.draw(canvas);
         }
 
-        int partialAlpha = (int) (alpha * normalized);
-        super.setAlpha(partialAlpha);
-        super.draw(canvas);
-        super.setAlpha(alpha);
+        if (useColorMatrixTransition) {
+          colorMatrix.reset();
+          TRANSITION.apply(colorMatrix, fraction);
+
+          // Scale the transition alpha to match the defined alpha on the drawable.
+          colorMatrix.getArray()[ROW_ALPHA + COLUMN_ALPHA] *= (alpha / 255);
+
+          super.setColorFilter(new ColorMatrixColorFilter(colorMatrix));
```

Comment:
I would suggest supporting this on L only and using `ColorMatrixColorFilter#setColorMatrix` via reflection to avoid allocations. On pre-L falling back to fade-in seems reasonable.

I'm aware this is not pretty, but currently I don't see other way of avoiding 13 `ColorMatrixColorFilter` for each image that gets faded in.